### PR TITLE
1 Set Up The Business Customer Relationship

### PR DESCRIPTION
**Pull request by leonardk1**
#1 
As A Developer
I Want To Set Up the Business Customer Relationship model
So That I can Be able to relate Businesses With Their customers

###### Tasks
- [x] create the relationship model
- [x] add configurations to the business model to have a `follower followed relationship` to be used in identifying customers and service providers
- [x] customers are the followers and service providers are the ones being followed

##### notes
in the business model note the following
- an `active_relationship` is the one where the current business is a customer to another
- a `passive_relationship` is one where the current business is a service provider to another business

### CODE REVIEW

**Comment by leonardk1** [comment]
@godfreym please review

File: spec/models/business_spec.rb [hunk 1]
```diff
@@ -41,4 +41,77 @@
       expect(@business.products.count).to be > 1
     end
   end
+
+  describe "adding and removing service providers" do
+    before do
+      Business.delete_all
+      @business3 = create(:business, name: "Food Supplies")
```
Comment (godfreym) [inline spec/models/business_spec.rb]: @leonardk1 why are you doing this business 1 or 2 or 3?, and why not create a factory which will create 8 or any number of names you need with different names

Comment (leonardk1) [inline spec/models/business_spec.rb]: @godfreym fixed

**Comment by leonardk1** [comment]
@arafatm please review

**Comment by leonardk1** [comment]
@godfreym checkout the changes please

File: spec/models/business_spec.rb (hunk 1 above)
Comment (godfreym) [inline spec/models/business_spec.rb]: @leonardk1 Looks good

**Comment by leonardk1** [comment]
@arafatm please review @godfreym please checkout the changes

**Comment by godfreym** [comment]
@leonardk1 Looks Good

**Comment by arafatm** [comment]
:+1:

**Pull request merged**